python/pyspark/sql/pandas/group_ops.py

```diff
@@ -505,22 +514,75 @@ def transformWithStateUDF(
 
             return result
 
-        if isinstance(outputStructType, str):
-            outputStructType = cast(StructType, _parse_datatype_string(outputStructType))
+        def transformWithStateWithInitStateUDF(
+                statefulProcessorApiClient: StatefulProcessorApiClient,
+                key: Any,
+                inputRows: Iterator["PandasDataFrameLike"],
+                # for non first batch, initialStates will be None
+                initialStates: Iterator["PandasDataFrameLike"] = None
+        ) -> Iterator["PandasDataFrameLike"]:
+            handle = StatefulProcessorHandle(statefulProcessorApiClient)
+
+            if statefulProcessorApiClient.handle_state == StatefulProcessorHandleState.CREATED:
```

Comment:
There's something not very clear to me here, could you help me understand more? 

We only call `handleInitialState` when handle state is `CREATED`, but after we processed the initial state of the first grouping key, we update the state to be `INITIALIZED`. Wouldn't that skip the initial state for other grouping keys?

Reply:
If my understanding is correct, we should move the `handleInitialState` outside the handle state check, do it after the `init` call.

Reply:
You are correct. I also moved out the code block and run a local test with partition number as "1" to confirm the implementation is correct.